docs/build/use-cases/key-generation.mdx

```diff
@@ -0,0 +1,576 @@
+
+## Install minimal deps
+
+We keep dependencies lean:
+
+```shell
+npm init -y
+npm i express ethers zod dotenv
+npm i -D typescript tsx @types/node @types/express hardhat
+npx tsc --init --rootDir src --outDir dist --module NodeNext --target ES2022
+```
+
+## App structure
+
+We'll add four small TS files and one Solidity contract:
+
+```
+src/
+  appd.ts        # talks to appd over /run/rofl-appd.sock
+  evm.ts         # ethers helpers (provider, wallet, tx)
+  keys.ts        # tiny helpers (checksum)
+  server.ts      # HTTP API to drive the demo
+contracts/
+  Counter.sol    # optional sample contract
+```
+
+### `src/appd.ts` — keygen over UNIX socket
+
+This calls `POST /rofl/v1/keys/generate` and returns a **0x‑hex private
+key**. Outside ROFL, it can fall back to a **dev key** for local tests.
+
+<details>
+  <summary>src/appd.ts</summary>
+
+```ts
+import { request } from "node:http";
+import { existsSync } from "node:fs";
+
+const APPD_SOCKET = "/run/rofl-appd.sock";
+type KeyKind = "secp256k1" | "ed25519" | "raw-256" | "raw-386";
+
+export async function generateKey(
+  keyId: string,
+  kind: KeyKind = "secp256k1"
+): Promise<string> {
+  if (!existsSync(APPD_SOCKET)) {
+    throw new Error("appd socket missing: /run/rofl-appd.sock");
+  }
+  const body = JSON.stringify({ key_id: keyId, kind });
+  return new Promise((resolve, reject) => {
+    const req = request(
+      {
+        method: "POST",
+        socketPath: APPD_SOCKET,
+        path: "/rofl/v1/keys/generate",
+        headers: {
+          "Content-Type": "application/json",
+          "Content-Length": Buffer.byteLength(body).toString()
+        }
+      },
+      (res) => {
+        let data = "";
+        res.setEncoding("utf8");
+        res.on("data", (c) => (data += c));
+        res.on("end", () => {
+          try {
+            const { key } = JSON.parse(data);
+            if (!key) throw new Error(`Bad response: ${data}`);
+            resolve(key.startsWith("0x") ? key : `0x${key}`);
+          } catch (e) {
+            reject(e);
+          }
+        });
+      }
+    );
+    req.on("error", reject);
+    req.write(body);
+    req.end();
+  });
+}
+
+export async function getAppId(): Promise<string> {
+  if (!existsSync(APPD_SOCKET)) {
+    throw new Error("appd socket missing: /run/rofl-appd.sock");
+  }
+  return new Promise((resolve, reject) => {
+    const req = request(
+      { method: "GET", socketPath: APPD_SOCKET, path: "/rofl/v1/app/id" },
+      (res) => {
+        let data = "";
+        res.setEncoding("utf8");
+        res.on("data", (c) => (data += c));
+        res.on("end", () => {
+          const id = data.trim();
+          if (!/^rofl1[0-9a-z]+$/.test(id)) {
+            return reject(new Error(`Bad app id: ${data}`));
+          }
+          resolve(id);
+        });
+      }
+    );
+    req.on("error", reject);
+    req.end();
+  });
+}
+
+export async function getEvmPrivateKey(keyId: string): Promise<string> {
```

Comment:
Follow best security practices and use `SecretKey` instead of `PrivateKey`.

```suggestion
export async function getEvmSecretKey(keyId: string): Promise<string> {
```